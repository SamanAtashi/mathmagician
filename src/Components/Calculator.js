import React, { Component } from 'react';
import Calculate from '../logic/calculate';
import './Calculator.css';

class Calculator extends Component {
  constructor() {
    super();
    this.state = {
      total: null,
      next: null,
      operation: null,
      display: 0,
    };
  }

    calcButton = (e) => {
      const btnName = e.target.innerHTML;
      const { total, next, operation } = Calculate(this.state, btnName);
      if (btnName === 'AC') {
        this.setState({
          display: 0, total, next, operation,
        });
      } else if (btnName === '=') {
        this.setState({
          display: total, total, next, operation,
        });
      } else if (btnName === '+/-') {
        this.setState({
          display: next || total, total, next, operation,
        });
      } else {
        this.setState((preState) => ({ display: btnName, ...Calculate(preState, btnName) }));
      }
    }

    render() {
      const list = [
        '1',
        '2',
        '3',
        '4',
        '5',
        '6',
        '7',
        '8',
        '9',
        'AC',
        '+/-',
        '%',
        'รท',
        '.',
        'x',
        '+',
        '-',
        '=',
        '0',
      ];

      const listItem = list.map((item, index) => (
        <li
          key={item.toString()}
          className={`calc-item o-${index + 1}`}
        >
          <button type="button" onClick={this.calcButton}>
            {item}
          </button>
        </li>
      ));

      const { display } = this.state;

      return (
        <ul className="d-flex">
          <li className="calc-result d-flex o-0">{display}</li>
          {listItem}
        </ul>
      );
    }
}

export default Calculator;
