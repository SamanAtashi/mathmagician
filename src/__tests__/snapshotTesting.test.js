import renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

describe('Snapshot testing components', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
