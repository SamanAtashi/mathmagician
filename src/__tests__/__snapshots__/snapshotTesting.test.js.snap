// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing components Calculator snapshot 1`] = `
<ul
  className="d-flex"
>
  <li
    className="calc-result d-flex o-0"
  >
    0
  </li>
  <li
    className="calc-item o-1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </li>
  <li
    className="calc-item o-2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
  </li>
  <li
    className="calc-item o-3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
  </li>
  <li
    className="calc-item o-4"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </li>
  <li
    className="calc-item o-5"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </li>
  <li
    className="calc-item o-6"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
  </li>
  <li
    className="calc-item o-7"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
  </li>
  <li
    className="calc-item o-8"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
  </li>
  <li
    className="calc-item o-9"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </li>
  <li
    className="calc-item o-10"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
  </li>
  <li
    className="calc-item o-11"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
  </li>
  <li
    className="calc-item o-12"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
  </li>
  <li
    className="calc-item o-13"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </li>
  <li
    className="calc-item o-14"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
  </li>
  <li
    className="calc-item o-15"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </li>
  <li
    className="calc-item o-16"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </li>
  <li
    className="calc-item o-17"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </li>
  <li
    className="calc-item o-18"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </li>
  <li
    className="calc-item o-19"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      0
    </button>
  </li>
</ul>
`;
